user/views.py

```diff
@@ -0,0 +1,103 @@
+import json
+import requests
+
+from django.views import View
+from django.http  import JsonResponse
+
+from my_settings import SECRET, ALGORITHM
+from .models     import User
+from core.utils  import get_hashed_pw, is_valid_name, is_valid_email, is_valid_password, checkpw, issue_token
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            name     = data['name']
+            email    = data['email']
+            password = data['password']
+
+            if User.objects.filter(email = email).exists():
+                return JsonResponse({'MESSAGE':'DUPLICATED_INFORMATION'}, status=400)
+
+            if not is_valid_name(name):
+                return JsonResponse({'MESSAGE':'INVALID_NAME'}, status=400)
+
+            if not is_valid_email(email):
+                return JsonResponse({'MESSAGE':'INVALID_EMAIL'}, status=400)
+
+            if not is_valid_password(password):
+                return JsonResponse({'MESSAGE':'INVALID_PASSWORD'}, status=400)
+
+            User.objects.create(
+                name     = name,
+                email    = email,
+                password = get_hashed_pw(password)
+            )
+
+            return JsonResponse({'MESSAGE':'SUCCESS'}, status=201)
+        except json.JSONDecodeError as e :
+            return JsonResponse({'MESSAGE': f'JSON_ERROR:{e}'}, status=400)
+        except TypeError :
+            return JsonResponse({'MESSAGE': 'TYPE_ERROR'}, status=400)
+        except KeyError as e :
+            return JsonResponse({'MESSAGE': f'KEY_ERROR:{e}'}, status=400)
+
+class LogInView(View):
+    def post(self, request):
+        try:
+            data     = json.loads(request.body)
+
+            email    = User.objects.filter(email=data['email'])
+            #라인 53 지우고 라인49의 filter를 get으로 하면 안되나요?
+            #없으면 에러핸들링에서 잡아줄 것 같은데,
+            password = data['password']
+            user     = User.objects.get(email=data['email'])
+
+            if not checkpw(password, user):
+                return JsonResponse({'MESSAGE':'INVALID_PASSWORD'}, status=400)
+
+            token = issue_token(user.id)
+
+            return JsonResponse({'token':token, 'name':user.name}, status=200)
+
+        except User.DoesNotExist : 
+            return JsonResponse({'MESSAGE': 'NO_EXIST_USER'}, status=401)
+        except json.JSONDecodeError as e :
+            return JsonResponse({'MESSAGE': f'JSON_ERROR:{e}'}, status=400)
+        except TypeError :
+            return JsonResponse({'MESSAGE': 'TYPE_ERROR'}, status=400)
+        except KeyError as e :
+            return JsonResponse({'MESSAGE': f'KEY_ERROR:{e}'}, status=400)
+
+
+class KakaoLogInView(View):
+    def post(self, request):
+        try:
+            access_token = request.headers['Authorization']
+            url          = 'https://kapi.kakao.com/v2/user/me'
+            header       = {
```

Comment:
```suggestion
            headers       = {
```
원래 `headers` 가 맞으니 여기도 헷갈리지 않게 `headers` 라고 선언해주세요.

Reply:
앗.. 오히려 headers = headers 하면 읽을 때 헷갈릴까바 뺏는데,, 오히려 똑같이 하는게 좋군요